contributivity.py

```diff
@@ -37,3 +93,966 @@ def __str__(self):
+                    mu[k, strata] = (
+                        mu[k, strata] * (length - 1) + increment
+                    ) / length
+                    S
+                    # computes the intra-strata standard deviation
+                    sigma2[k, strata] = 0 
+                    for v in increments_generated[k][strata].values():
+                        sigma2[k, strata] +=   (v- mu[k, strata]) ** 2
+                    if length > 1: 
+                        sigma2[k, strata] /= length - 1
+                    else: #avoid creating a Nan value  when length = 1 
+                        sigma2[k, strata]=0
+                    sigma2[k, strata]*= (1/length-   factorial(N - 1 - strata) * factorial(strata)  / factorial(N-1) )
+                    print("t : ",t,",k :",k ,", strata :",strata, ", sigma2 :",sigma2[k])
+                shap = np.mean(mu, axis=0)
+                var = np.zeros(N)  # variance of the estimator
+                for k in range(N):
+                    for strata in range(N): 
+                        n_k_strata = len (increments_generated[k][strata])
+                        # compute the variance of the estimator times N**2
+                        if n_k_strata == 0:
+                            var[k] = np.Inf
+                        else:
+                            var[k] += sigma2[k, strata] ** 2 / n_k_strata
+                        # handle the while condition and the next allocations 
+                        ## if the number of allocation is above 20 in each strat we can stop
+                        if n_k_strata > 20 : 
+                            continuer[k][strata] = False
+                        ## if a strata as been fully explored we stop allocating to this strata
+                        if  len(increments_generated[k][strata]) ==  factorial(N-1)/ (factorial(N - 1 - strata) * factorial(strata) ) :
+                            continuer[k][strata] = False 
+                    var[k] /= N ** 2 #correct the variance of the estimator
+                v_max = np.max(var)
+            self.name = "WR_SMC Shapley values"
+            self.contributivity_scores = shap
+            self.scores_std = np.sqrt(var)
+            self.normalized_scores = self.contributivity_scores / np.sum(
+                self.contributivity_scores
+            )
+            end = timer()
+            self.computation_time = end - start                        
+        
+    def compute_contributivity(self, method_to_compute, 
+                               current_scenario,
+                               sv_accuracy=0.01, 
+                               alpha=0.95, 
+                               truncation=0.05,
+                               update=50):
+        
+        if method_to_compute == "Shapley values":
+            # Contributivity 1: Baseline contributivity measurement (Shapley Value)
+            self.compute_SV(current_scenario) 
+        elif method_to_compute == "Independant scores raws":
```

Comment:
raw without S

Reply:
After discussing it with Eric, I remove the additive scores, so we can call it "independent scores"